tests/functional/functional_test.py

```diff
@@ -16,7 +16,15 @@ def setUp(self):
         return self
 
     def tearDown(self):
-        shutil.rmtree(self.__path)
+        def rmtree_onerror(func, path, exc_info):
+            ''' from http://stackoverflow.com/questions/2656322/shutil-rmtree-fails-on-windows-with-access-is-denied '''
+            import stat
+            if not os.access(path, os.W_OK):
+                os.chmod(path, stat.S_IWUSR)
+                func(path)
+            else:
+                raise
+        shutil.rmtree(self.__path, onerror=rmtree_onerror)
```

Comment:
can you move this to `filesystem`? maybe as a `force_rmtree` function? definitely gonna have some use-cases for this soon